include/deal.II/lac/trilinos_tpetra_solver.h

```diff
@@ -0,0 +1,227 @@
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_trilinos_tpetra_solver_h
+#define dealii_trilinos_tpetra_solver_h
+
+
+#include <deal.II/base/config.h>
+
+#include "deal.II/base/types.h"
+
+// #include <Tpetra_KokkosCompat_DefaultNode.hpp>
+// #include <Tpetra_Operator.hpp>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  include <deal.II/lac/la_parallel_vector.h>
+#  include <deal.II/lac/solver_control.h>
+#  include <deal.II/lac/trilinos_tpetra_vector.h>
+
+#  include <Amesos2.hpp>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  namespace TpetraWrappers
+  {
+    // forward declarations
+#  ifndef DOXYGEN
+    template <typename Number, typename Node>
+    class SparseMatrix;
+#  endif
+
+    /**
+     * An implementation of Trilinos Tpetra direct solvers (using the Amesos2
+     * package). The data field AdditionalData::solver_type can be used to
+     * specify the type of solver. It allows the use of built-in solvers KLU2 as
+     * well as third-party solvers SuperLU_DIT or MUMPS.
+     *
+     * For instructions on how to install Trilinos for use with direct solvers
+     * other than KLU, see the link to the Trilinos installation instructions
+     * linked to from the deal.II ReadMe file.
+     *
+     * @ingroup TpetraWrappers
+     */
+    template <
+      typename Number,
+      typename Node = Tpetra::KokkosClassic::DefaultNode::DefaultNodeType,
+      typename LinearOperator = Tpetra::
+        Operator<Number, int, dealii::types::signed_global_dof_index, Node>,
+      typename MultiVector = Tpetra::
+        Operator<Number, int, dealii::types::signed_global_dof_index, Node>>
+    class SolverDirect
+    {
+    public:
+      /**
+       * Standardized data struct to pipe additional data to the solver.
+       */
+
+      struct AdditionalData
+      {
+        /**
+         * Set the additional data field to the desired output format.
+         */
+        explicit AdditionalData(const bool output_solver_details = false,
+                                const std::string &solver_type   = "KLU2");
+
+        /**
+         * Enables/disables the output of solver details (residual in each
+         * iterations etc.).
+         */
+        bool output_solver_details;
+
+        /**
+         * Set the solver type (for third party solver support of Trilinos
+         * Amesos package). Possibilities are:
+         * <ul>
+         * <li>  "LAPACK" </li>
+         * <li>  "ScaLAPACK" </li>
+         * <li>  "KLU2" </li>
+         * <li>  "UMFPACK" </li>
+         * <li>  "Pardiso" </li>
+         * <li>  "SuperLU" </li>
+         * <li>  "SuperLU_DIST" </li>
+         * <li>  "MUMPS" </li>
+         * </ul>
```

Comment:
Is this a complete list of what Amesos supports (I think you actually mean to say Amesos2)? I thought there is also ShyLU and Basker.

Reply:
No there is a bunch more and we can list them all here because Amesos2::query(solver_name)
allows for an easy check on whether the solver is supported or not. 

I was not aware of that when writing the original list and will add all possibilities to the header file.

Reply:
The list is now complete as far as I can see it.